apps/overige-objecten-api/src/controllers/objects/index.ts

```diff
@@ -1,11 +1,24 @@
 import type { RequestHandler } from 'express';
 import { GET_ALL_PRODUCTS, GET_ALL_VAC_ITEMS, GET_PRODUCT_BY_UUID, GET_VAC_ITEM_BY_UUID } from '../../queries';
 import type { StrapiProductType, VACSData } from '../../strapi-product-type';
+import type { components } from '../../types/openapi';
 import { fetchData, generateKennisartikelObject, getPaginatedResponse, getTheServerURL } from '../../utils';
+
+type GetKennisartikelReturnData = components['schemas']['ObjectData'];
 export const getAllObjectsController: RequestHandler = async (req, res, next) => {
   try {
     const locale = req.query?.locale || 'nl';
     const type = (req.query?.type as string) || '';
+    const typeUrl = req.query?.type ? new URL(req.query?.type as string) : '';
+    const isURL = typeof typeUrl === 'object';
+    const isVac = isURL && typeUrl.pathname.split('/').includes('vac');
+    const isKennisartikel = isURL && typeUrl.pathname.split('/').includes('kennisartikel');
+    // If no specific type, return empty array
+    if (type && !isVac && !isKennisartikel) {
+      return res.status(200).json({
+        results: [],
+      });
```

Comment:
Ik vraag me af: zou het verwacht gedrag zijn dat de API altijd pagination teruggeeft, ook als het maar 1 pagina is met 0 resultaten?